main.py

```diff
@@ -5031,6 +5026,102 @@ def do_command(self, command, args, sender, channel, tags, isWhisper=False):
+                        
+                        try:
+                            validateImageURL(args[2])
+                        except ValueError as ex:
+                            self.message(channel, "Invalid link specified. %s" % str(ex), isWhisper)
+                            return
+
+                        if canManageImages:
+                            # automatically do the change
+                            try:
+                                hostedURL = processImageURL(args[2])
+                            except Exception as ex:
+                                self.message(channel, "Could not process image. %s" % str(ex), isWhisper)
+                                return
+                            
+                            if do_global:
+                                cur.execute("UPDATE waifus SET image = %s WHERE id = %s", [hostedURL, waifuid])
+                            else:
+                                cur.execute("UPDATE has_waifu SET custom_image = %s WHERE waifuid = %s AND userid = %s AND rarity = %s", [hostedURL, tags['user-id'], waifuid, godRarity])
+
+                            # log the change for posterity
+                            insertArgs = [tags['user-id'], waifuid, args[2], do_global, tags['user-id'], current_milli_time()]
+                            cur.execute("INSERT INTO godimage_requests (requesterid, waifuid, image, is_global, state, moderatorid, created) VALUES(%s, %s, %s, %s, 'auto_accepted', %s, %s)", insertArgs)
+
+                            self.message(channel, "Image change processed successfully.", isWhisper)
+                            return
+                        else:
+                            # cancel any old pending requests for this waifu
+                            cur.execute("UPDATE godimage_requests SET state = 'cancelled', updated = %s WHERE waifuid = %s AND state = 'pending'", [current_milli_time(), waifuid])
+
+                            # record a new request
+                            insertArgs = [tags['user-id'], waifuid, args[2], do_global, current_milli_time()]
+                            cur.execute("INSERT INTO godimage_requests (requesterid, waifuid, image, is_global, state, created) VALUES(%s, %s, %s, %s, 'pending', %s)", insertArgs)
+
+                            # notify the discordhook of the new request
+                            waifuInfo = getWaifuById(waifuid)
+                            discordArgs = {"user": tags['display-name'], "id": waifuid, "name": waifuInfo["name"], "image": args[2], "type": "a global" if do_global else "an"}
+                            discordbody = {
+                                "username": "WTCG Admin", 
+                                "content" : "{user} requested {type} image change for [{id}] {name} to <{image}>!\nUse `!godimage check {id}` in any chat to check it.".format(**discordArgs)
+                            }
+                            sendAdminDiscordAlert(discordbody)
```

Comment:
Given the regular `sendDiscordAlert` function is in a separate thread, is there any reason you're *not* putting sendAdminDiscordAlert in its own thread? Unless I misunderstand how the requests module works, the bot could hang here if the Discord API is unreachable.

Reply:
Oversight, actually - this should definitely be threaded, will fix later